data/builder/file.go

```diff
@@ -57,6 +58,91 @@ func BuildUnixFSFile(r io.Reader, chunker string, ls *ipld.LinkSystem) (ipld.Lin
 	}
 }
 
+// EstimateUnixFSFile estimates the byte size of the car file that would be
+// needed to hold a UnixFS file containing data of the given length.
+func EstimateUnixFSFileDefaultChunking(dataLength uint64) uint64 {
+	blkSize := chunk.DefaultBlockSize
+	blocks := dataLength / uint64(blkSize)
+	remainder := dataLength % uint64(blkSize)
+
+	size := dataLength
+	cidExample, _ := leafLinkProto.Prefix.Sum([]byte{0})
+	cidLength := uint64(len(cidExample.Bytes()))
+
+	links := []uint64{}
+	for i := uint64(0); i < blocks; i++ {
+		links = append(links, uint64(chunk.DefaultBlockSize))
+	}
+	// account for the uvarint + cid length of each block of raw data.
+	size += uint64(len(links)) * (cidLength + uint64(varint.UvarintSize(cidLength+uint64(blkSize))))
+	if remainder > 0 {
+		links = append(links, remainder)
+		size += cidLength + uint64(varint.UvarintSize(cidLength+uint64(remainder)))
+	}
+
+	// account for the metadata overhead nodes.
+	ls := cidlink.DefaultLinkSystem()
+	storage := cidlink.Memory{}
+	ls.StorageReadOpener = storage.OpenRead
+	ls.StorageWriteOpener = storage.OpenWrite
+
+	icnt := 0
+	for len(links) > 1 {
```

Comment:
Would it make sense to update the godoc of this function with the cost complexity of estimating size vs writing the CAR out into a temporary file?

Reply:
i think the extent of what we can really say is that the size calculation doesn't need memory, but will do the CPU work for generation of intermediate blocks.